rest/src/main/java/com/netflix/conductor/rest/controllers/TaskResource.java

```diff
@@ -83,6 +83,25 @@ public String updateTask(@RequestBody TaskResult taskResult) {
         return taskService.updateTask(taskResult);
     }
 
+    @PostMapping(value = "/{workflowId}/{taskRefName}/{status}", produces = TEXT_PLAIN_VALUE)
+    @Operation(summary = "Update a task By Ref Name")
+    public String updateTask(
+            @PathVariable("workflowId") String workflowId,
+            @PathVariable("taskRefName") String taskRefName,
+            @PathVariable("status") TaskResult.Status status,
+            @RequestBody Map<String, Object> output) {
+
+        Task pending = taskService.getPendingTaskForWorkflow(workflowId, taskRefName);
```

Comment:
In the current implementation, this is an expensive DB operation and the DB queries are not optimized to serve this well.
Additionally, the API itself is designed to be too constrained to only allow updating status and output. It should ideally allow updating the TaskResult.

Having said that, we are working on a proposal to overhaul the DB schema model to better support large workflows and enable support for these types of queries. We will revisit this API at that time.